client/lib/signup/flow-controller.js

```diff
@@ -84,10 +86,17 @@ assign( SignupFlowController.prototype, {
 				return true;
 			}
 
-			dependenciesNotProvided = difference( step.dependencies, this._getFlowProvidesDependencies() );
+			const dependenciesFound = pick( SignupDependencyStore.get(), step.dependencies );
+			dependenciesNotProvided = difference( step.dependencies, keys( dependenciesFound ), this._getFlowProvidesDependencies() );
 			if ( ! isEmpty( dependenciesNotProvided ) ) {
+
+				if ( this._flowName !== flows.defaultFlowName ) {
+					// redirect to the default signup flow, hopefully it will be valid
+					page( utils.getStepUrl() );
```

Comment:
I feel like it's good to still throw the error, because these unhandled errors are logged to tracks.